In [1]:
import os
from tqdm import tqdm
import pandas as pd

---

### 讀取資料夾的方法

In [2]:
#10個資料夾間需要比對的次數
9+8+7+6+5+4+3+2+1

45

率性休閒 E1_1
簡約質感 E1_2
清新甜美 E1_3
都會輕熟 E1_4
溫柔浪漫 E1_5
慵懶度假 E1_6
大氣優雅 E1_7
性感迷人 E1_8
態度個性 E1_9
其他 E1_10

In [29]:
style_dir = os.listdir("素材風格(主圖)")
style_dir.remove(".DS_Store")
style_dir


['其他 E1_10',
 '率性休閒 E1_1',
 '性感迷人 E1_8',
 '態度個性 E1_9',
 '清新甜美 E1_3',
 '都會輕熟 E1_4',
 '慵懶度假 E1_6',
 '簡約質感 E1_2',
 '溫柔浪漫 E1_5',
 '大氣優雅 E1_7']

In [56]:
style_dir = os.listdir("素材風格(主圖)")
style_dir.remove(".DS_Store")
mutual_df = pd.DataFrame(columns=['style_dirs','mutual_pics'])

for i in style_dir:
    first = os.listdir("素材風格(主圖)/"+i)
    no = style_dir.index(i) + 1
    second_list = style_dir[no:]
    for j in second_list:
        second = os.listdir("素材風格(主圖)/"+j)
        mutual = [x for x in first for y in second if x == y]
        dirs = str(i)+" 與 "+str(j)
        new_row = {'style_dirs': dirs, 'mutual_pics': len(mutual)}
        mutual_df.loc[len(mutual_df)] =  new_row
    



In [57]:
mutual_df

style_dirs mutual_pics
0    其他 E1_10 與 率性休閒 E1_1          93
1    其他 E1_10 與 性感迷人 E1_8          12
2    其他 E1_10 與 態度個性 E1_9          57
3    其他 E1_10 與 清新甜美 E1_3           8
4    其他 E1_10 與 都會輕熟 E1_4           8
5    其他 E1_10 與 慵懶度假 E1_6           2
6    其他 E1_10 與 簡約質感 E1_2          25
7    其他 E1_10 與 溫柔浪漫 E1_5           1
8    其他 E1_10 與 大氣優雅 E1_7           6
9   率性休閒 E1_1 與 性感迷人 E1_8         105
10  率性休閒 E1_1 與 態度個性 E1_9         970
11  率性休閒 E1_1 與 清新甜美 E1_3         489
12  率性休閒 E1_1 與 都會輕熟 E1_4          96
13  率性休閒 E1_1 與 慵懶度假 E1_6          94
14  率性休閒 E1_1 與 簡約質感 E1_2         572
15  率性休閒 E1_1 與 溫柔浪漫 E1_5          29
16  率性休閒 E1_1 與 大氣優雅 E1_7          46
17  性感迷人 E1_8 與 態度個性 E1_9          65
18  性感迷人 E1_8 與 清新甜美 E1_3          16
19  性感迷人 E1_8 與 都會輕熟 E1_4          14
20  性感迷人 E1_8 與 慵懶度假 E1_6          34
21  性感迷人 E1_8 與 簡約質感 E1_2          79
22  性感迷人 E1_8 與 溫柔浪漫 E1_5          18
23  性感迷人 E1_8 與 大氣優雅 E1_7          30
24  態度個性 E1_9 與 清新甜美 E1_3          42
25  態度個性 E1_9 與 都會輕熟 E1_4          33
26  態度個性 E1_9 與 慵懶度假 E1_6          38
27  態度個性 E1_9 與 簡約質感 E1_2         151
28  態度個性 E1_9 與 溫柔浪漫 E1_5           1
29  態度個性 E1_9 與 大氣優雅 E1_7          14
30  清新甜美 E1_3 與 都會輕熟 E1_4           6
31  清新甜美 E1_3 與 慵懶度假 E1_6         100
32  清新甜美 E1_3 與 簡約質感 E1_2         145
33  清新甜美 E1_3 與 溫柔浪漫 E1_5         831
34  清新甜美 E1_3 與 大氣優雅 E1_7          40
35  都會輕熟 E1_4 與 慵懶度假 E1_6           8
36  都會輕熟 E1_4 與 簡約質感 E1_2         243
37  都會輕熟 E1_4 與 溫柔浪漫 E1_5          39
38  都會輕熟 E1_4 與 大氣優雅 E1_7         144
39  慵懶度假 E1_6 與 簡約質感 E1_2         131
40  慵懶度假 E1_6 與 溫柔浪漫 E1_5          42
41  慵懶度假 E1_6 與 大氣優雅 E1_7          36
42  簡約質感 E1_2 與 溫柔浪漫 E1_5         145
43  簡約質感 E1_2 與 大氣優雅 E1_7         557
44  溫柔浪漫 E1_5 與 大氣優雅 E1_7         210

In [70]:
mutual_df = mutual_df.sort_values(['mutual_pics'],ascending=False)
mutual_df.head(10)

style_dirs mutual_pics
10  率性休閒 E1_1 與 態度個性 E1_9         970
33  清新甜美 E1_3 與 溫柔浪漫 E1_5         831
14  率性休閒 E1_1 與 簡約質感 E1_2         572
43  簡約質感 E1_2 與 大氣優雅 E1_7         557
11  率性休閒 E1_1 與 清新甜美 E1_3         489
36  都會輕熟 E1_4 與 簡約質感 E1_2         243
44  溫柔浪漫 E1_5 與 大氣優雅 E1_7         210
27  態度個性 E1_9 與 簡約質感 E1_2         151
32  清新甜美 E1_3 與 簡約質感 E1_2         145
42  簡約質感 E1_2 與 溫柔浪漫 E1_5         145

---

### 讀取csv的方法

In [4]:
s_df = pd.read_excel("P1圖對應素材風格(主圖).xlsx")
s_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9746 entries, 0 to 9745
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   主要系列編號     9746 non-null   object
 1   GroupName  9746 non-null   object
 2   TagCode    9746 non-null   object
 3   TagName    9746 non-null   object
 4   TypeName   9746 non-null   object
 5   P1圖圖片連結    9746 non-null   object
dtypes: object(6)
memory usage: 457.0+ KB


In [35]:
s1 = s_df[s_df['TagCode']=="E1_1"]
s1_df = pd.DataFrame(s1.主要系列編號)
s1_df = s1_df.rename({'主要系列編號':'E1_1'},axis=1)

s2 = s_df[s_df['TagCode']=="E1_2"]
s2_df = pd.DataFrame(s2.主要系列編號)
s2_df = s2_df.rename({'主要系列編號':'E1_2'},axis=1)

s3 = s_df[s_df['TagCode']=="E1_3"]
s3_df = pd.DataFrame(s3.主要系列編號)
s3_df = s3_df.rename({'主要系列編號':'E1_3'},axis=1)

s4 = s_df[s_df['TagCode']=="E1_4"]
s4_df = pd.DataFrame(s4.主要系列編號)
s4_df = s4_df.rename({'主要系列編號':'E1_4'},axis=1)

s5 = s_df[s_df['TagCode']=="E1_5"]
s5_df = pd.DataFrame(s5.主要系列編號)
s5_df = s5_df.rename({'主要系列編號':'E1_5'},axis=1)

s6 = s_df[s_df['TagCode']=="E1_6"]
s6_df = pd.DataFrame(s6.主要系列編號)
s6_df = s6_df.rename({'主要系列編號':'E1_6'},axis=1)

s7 = s_df[s_df['TagCode']=="E1_7"]
s7_df = pd.DataFrame(s7.主要系列編號)
s7_df = s7_df.rename({'主要系列編號':'E1_7'},axis=1)

s8 = s_df[s_df['TagCode']=="E1_8"]
s8_df = pd.DataFrame(s8.主要系列編號)
s8_df = s8_df.rename({'主要系列編號':'E1_8'},axis=1)

s9 = s_df[s_df['TagCode']=="E1_9"]
s9_df = pd.DataFrame(s9.主要系列編號)
s9_df = s9_df.rename({'主要系列編號':'E1_9'},axis=1)

s10 = s_df[s_df['TagCode']=="E1_10"]
s10_df = pd.DataFrame(s10.主要系列編號)
s10_df = s10_df.rename({'主要系列編號':'E1_10'},axis=1)

df = pd.concat([s1_df,s2_df,s3_df,s4_df,s5_df,s6_df,s7_df,s8_df,s9_df,s10_df], axis=1)

In [4]:
columns = ['E1_1','E1_2','E1_3','E1_4','E1_5','E1_6','E1_7','E1_8','E1_9','E1_10']
index = ['E1_1','E1_2','E1_3','E1_4','E1_5','E1_6','E1_7','E1_8','E1_9','E1_10']
m_df = pd.DataFrame(columns=columns,index=index)
m_df

E1_1 E1_2 E1_3 E1_4 E1_5 E1_6 E1_7 E1_8 E1_9 E1_10
E1_1   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_2   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_3   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_4   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_5   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_6   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_7   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_8   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_9   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
E1_10  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN

In [44]:
s_df = pd.read_excel("P1圖對應素材風格(主圖).xlsx")
c_list = m_df.columns.to_list()
i_list = m_df.index.to_list()
for column in tqdm(c_list):
    first = s_df[s_df['TagCode']== column].主要系列編號
    c = c_list.index(column)
    for index in i_list:
        second = s_df[s_df['TagCode']== index].主要系列編號
        mutual = [x for x in first for y in second if x == y]
        i = i_list.index(index)
        m_df.iloc[i,c] = len(mutual)


100%|██████████| 10/10 [00:11<00:00,  1.11s/it]


In [45]:
m_df

E1_1  E1_2  E1_3 E1_4  E1_5 E1_6 E1_7 E1_8  E1_9 E1_10
E1_1   2304   588   494   98    30  102   52  106   974    94
E1_2    588  1710   148  257   150  140  576   82   156    26
E1_3    494   148  1687    7   831  106   41   16    42     9
E1_4     98   257     7  365    39    8  155   16    33     9
E1_5     30   150   831   39  1175   44  211   18     1     1
E1_6    102   140   106    8    44  367   37   34    40     2
E1_7     52   576    41  155   211   37  831   32    15     9
E1_8    106    82    16   16    18   34   32  246    65    13
E1_9    974   156    42   33     1   40   15   65  1026    57
E1_10    94    26     9    9     1    2    9   13    57   123

In [48]:
m_df.style.highlight_max(color='red',axis=0)

---

### 合併重疊風格的話 重新比對重複性

前5大互相重複的風格

- 率性休閒 E1_1(2304筆) 與 態度個性 E1_9(1026筆)	970
- 清新甜美 E1_3(1687筆) 與 溫柔浪漫 E1_5(1175筆)	831
- 率性休閒 E1_1(2304筆) 與 簡約質感 E1_2(1710筆)	572
- 簡約質感 E1_2(1710筆) 與 大氣優雅 E1_7(831筆)	557
- 率性休閒 E1_1(2304筆) 與 清新甜美 E1_3(1687筆)	489

In [46]:
s_df = pd.read_excel("P1圖對應素材風格(主圖).xlsx")

In [47]:
s_df['TagCode'] = s_df['TagCode'].replace({"E1_1":"E1_1_9","E1_9":"E1_1_9","E1_3":"E1_3_5","E1_5":"E1_3_5"})

In [34]:
set(s_df['TagCode'])

{'E1_10', 'E1_1_9', 'E1_2', 'E1_3_5', 'E1_4', 'E1_6', 'E1_7', 'E1_8'}

In [48]:
#合併風格後要去重複值
s_df.drop(s_df[s_df.duplicated(subset=['主要系列編號','TagCode'])].index,inplace=True)

In [49]:
s_df

主要系列編號 GroupName TagCode TagName TypeName  \
0     AA4231      素材風格  E1_3_5    清新甜美      P1圖   
2     AA7152      素材風格  E1_1_9    率性休閒      P1圖   
4     AA8469      素材風格  E1_3_5    溫柔浪漫      P1圖   
5     AA8469      素材風格    E1_7    大氣優雅      P1圖   
6     AA8750      素材風格  E1_3_5    清新甜美      P1圖   
...      ...       ...     ...     ...      ...   
9741  ZQ0041      素材風格  E1_1_9    率性休閒      P1圖   
9742  ZQ0042      素材風格  E1_1_9    率性休閒      P1圖   
9743  ZQ0045      素材風格  E1_1_9    率性休閒      P1圖   
9744  ZQ0046      素材風格  E1_1_9    率性休閒      P1圖   
9745  ZQ0047      素材風格  E1_1_9    率性休閒      P1圖   

                                                P1圖圖片連結  
0     https://image.obdesign.com.tw/catalog/products...  
2     https://image.obdesign.com.tw/catalog/products...  
4     https://image.obdesign.com.tw/catalog/products...  
5     https://image.obdesign.com.tw/catalog/products...  
6     https://image.obdesign.com.tw/catalog/products...  
...                                                 ...  
9741  https://image.obdesign.com.tw/catalog/products...  
9742  https://image.obdesign.com.tw/catalog/products...  
9743  https://image.obdesign.com.tw/catalog/products...  
9744  https://image.obdesign.com.tw/catalog/products...  
9745  https://image.obdesign.com.tw/catalog/products...  

[7902 rows x 6 columns]

In [56]:
for i in s_df['主要系列編號'].loc[s_df['TagCode']=="E1_7"]:

1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1


In [61]:
mutual_df = pd.DataFrame(columns=['style_type','mutual_pics'])
style_list = list(set(s_df['TagCode']))
for i in tqdm(style_list):
    first = s_df['主要系列編號'].loc[s_df['TagCode']== i ]
    no = style_list.index(i) + 1
    second_list = style_list[no:]
    for j in second_list:
        second = s_df['主要系列編號'].loc[s_df['TagCode']== j ]
        mutual = [x for x in first for y in second if x == y]
        dirs = str(i)+" 與 "+str(j)
        new_row = {'style_type': dirs, 'mutual_pics': len(mutual)}
        mutual_df.loc[len(mutual_df)] =  new_row

100%|██████████| 8/8 [00:04<00:00,  1.98it/s]


In [66]:
mutual_df

style_type mutual_pics
0       E1_4 與 E1_6           8
1      E1_4 與 E1_10           8
2       E1_4 與 E1_8          14
3     E1_4 與 E1_3_5          45
4     E1_4 與 E1_1_9          98
5       E1_4 與 E1_7         144
6       E1_4 與 E1_2         243
7      E1_6 與 E1_10           2
8       E1_6 與 E1_8          34
9     E1_6 與 E1_3_5         134
10    E1_6 與 E1_1_9         101
11      E1_6 與 E1_7          36
12      E1_6 與 E1_2         131
13     E1_10 與 E1_8          12
14   E1_10 與 E1_3_5           9
15   E1_10 與 E1_1_9          94
16     E1_10 與 E1_7           6
17     E1_10 與 E1_2          25
18    E1_8 與 E1_3_5          30
19    E1_8 與 E1_1_9         115
20      E1_8 與 E1_7          30
21      E1_8 與 E1_2          79
22  E1_3_5 與 E1_1_9         507
23    E1_3_5 與 E1_7         234
24    E1_3_5 與 E1_2         271
25    E1_1_9 與 E1_7          50
26    E1_1_9 與 E1_2         605
27      E1_7 與 E1_2         555

In [67]:
mutual_df.sort_values(['mutual_pics'],ascending=False)


style_type mutual_pics
26    E1_1_9 與 E1_2         605
27      E1_7 與 E1_2         555
22  E1_3_5 與 E1_1_9         507
24    E1_3_5 與 E1_2         271
6       E1_4 與 E1_2         243
23    E1_3_5 與 E1_7         234
5       E1_4 與 E1_7         144
9     E1_6 與 E1_3_5         134
12      E1_6 與 E1_2         131
19    E1_8 與 E1_1_9         115
10    E1_6 與 E1_1_9         101
4     E1_4 與 E1_1_9          98
15   E1_10 與 E1_1_9          94
21      E1_8 與 E1_2          79
25    E1_1_9 與 E1_7          50
3     E1_4 與 E1_3_5          45
11      E1_6 與 E1_7          36
8       E1_6 與 E1_8          34
18    E1_8 與 E1_3_5          30
20      E1_8 與 E1_7          30
17     E1_10 與 E1_2          25
2       E1_4 與 E1_8          14
13     E1_10 與 E1_8          12
14   E1_10 與 E1_3_5           9
1      E1_4 與 E1_10           8
0       E1_4 與 E1_6           8
16     E1_10 與 E1_7           6
7      E1_6 與 E1_10           2

In [73]:
a = len(s_df[s_df['TagCode'] == "E1_1_9"])
b = len(s_df[s_df['TagCode'] == "E1_2"])
c = len(s_df[s_df['TagCode'] == "E1_7"])
d = len(s_df[s_df['TagCode'] == "E1_3_5"])

print(f'E1_1_9 有{a}筆，E1_2有{b}筆，兩個重疊605筆')
print(f'E1_7 有{c}筆，E1_2有{b}筆，兩個重疊555筆')
print(f'E1_1_9 有{a}筆，E1_3_5有{d}筆，兩個重疊605筆')


E1_1_9 有2333筆，E1_2有1663筆，兩個重疊605筆
E1_7 有813筆，E1_2有1663筆，兩個重疊555筆
E1_1_9 有2333筆，E1_3_5有2019筆，兩個重疊605筆
